ocpmodels/trainers/base_trainer.py

```diff
@@ -403,24 +403,25 @@ def load_checkpoint(self, checkpoint_path):
         self.best_val_metric = checkpoint.get("best_val_metric", None)
         self.primary_metric = checkpoint.get("primary_metric", None)
 
-        # Load model, optimizer, normalizer state dict.
-        # if trained with ddp and want to load in non-ddp, modify keys from
-        # module.module.. -> module..
-        first_key = next(iter(checkpoint["state_dict"]))
-        if (
-            not distutils.initialized() or self.config["noddp"]
-        ) and first_key.split(".")[1] == "module":
-            # No need for OrderedDict since dictionaries are technically ordered
-            # since Python 3.6 and officially ordered since Python 3.7
-            new_dict = {k[7:]: v for k, v in checkpoint["state_dict"].items()}
-            self.model.load_state_dict(new_dict)
-        elif distutils.initialized() and first_key.split(".")[1] != "module":
+        ckpt_key_count = next(iter(checkpoint["state_dict"])).count("module")
+        mod_key_count = next(iter(self.model.state_dict())).count("module")
+        key_count_diff = mod_key_count - ckpt_key_count
+
+        if key_count_diff > 0:
+            new_dict = {
+                key_count_diff * "module." + k: v
+                for k, v in checkpoint["state_dict"].items()
+            }
+        elif key_count_diff < 0:
             new_dict = {
-                f"module.{k}": v for k, v in checkpoint["state_dict"].items()
+                k[7 * abs(key_count_diff) :]: v
+                for k, v in checkpoint["state_dict"].items()
             }
-            self.model.load_state_dict(new_dict)
         else:
-            self.model.load_state_dict(checkpoint["state_dict"])
+            new_dict = checkpoint["state_dict"]
+
```

Comment:
This logic is missing the case when the combination is consistent - `key_count_diff==0`?